tests/cache/tests.py

```diff
@@ -89,6 +90,25 @@ def empty_response(request):
 )
 
 
+def retry(retries=3, delay=1):
+    def decorator(func):
```

Comment:
Nitpick: shall we ensure that retries is > 0 for this helper to make sense?

Reply:
no, I guess? we could have something like:
```python
    if not isinstance(retries, int) or retries < 1:
        raise ValueError("Retries must be a positive integer.")
    if not isinstance(delay, (int, float)) or delay < 0:
        raise ValueError("Delay must be a non-negative number.")
```
But,
I don't see a use case where the retries number might be set by a function result or a config file somewhere... it will always be a hardcoded value by someone, and I really doubt someone will set negative retries. ...

Reply:
for providing a retry decorator that works with methods or a whole class,

we could either subclass **test.utils.TestContextDecorator** and somehow fill in enable/disable

or like:
```py
def retry(retries=3, delay=1):
    def decorator(obj):
        if issubclass(cls, TestCase):  # If it's a test class
            # Iterate over all attributes
            for attr_name, attr_value in obj.__dict__.items():
                # Apply the decorator to instance methods only, excluding special methods			
                if callable(attr_value) and not attr_name.startswith("__"):
                    setattr(obj, attr_name, decorator(attr_value))
					# maybe even make sure they start with "test_**"
            return obj
        else:  # If it's a function or method
            @wraps(obj)
            def wrapper(*args, **kwargs):
                attempts = 0
                ##### the rest here 
				##### call iscoroutinefunction and handle both async/sync
            return wrapper
    return decorator

# Example usage:
@retry(retries=5, delay=2)
class TestThingy(TestCase):
```   
-------

@nessita 

implementation aside I really don't think we should go there, Ideally we want tests to avoid using this decorator, 
we don't want to make it easier to write flaky tests, but only mitigate the existing?

Reply:
Thank you @wassef911 for sharing your view!

Overall I agree, the idea is not for this decorator to be used often. My worry is that these cache tests classes are showing flakiness in various tests, in an unpredictable manner. So for this file alone, I think that being able to decorate per class could be handy.

With that in mind, I think it's a fair request to progress this as is and later evaluate the need of a class-based decorator. I'll progress the review with that in mind.